docs/support/troubleshooting-hanging-tasks.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Troubleshooting Failed or Hanging Tasks
+description: A guide for troubleshooting tasks that fail unexpectedly or are unresponsive
+position: 8
+---
+
+Sometimes your deployments, health checks, or other tasks may unexpectedly fail, or can even appear unresponsive. This page describes some common issues along with strategies to help you overcome them.
+
+## Check the Logs
+
+The first step to debug your failed tasks is to check the [Task Log](docs/support/get-the-raw-output-from-a-task.md). This will usually contain detailed information about the failure. For deployments, this includes the step, and information about the deployment targets that the step was running on.
+
+If a deployment failed unexpectedly within a built-in step, you may have misconfigured the step. Double check the configuration of your step in your deployment process. If your step is relying on variables, then you may have also misconfigured your variables. There are [some methods](docs/support/debug-problems-with-octopus-variables.md) available that can help you debug your variables.
+
+If none of the above steps help, then you may have encountered a bug in a built-in step, in which case you can contact support for further assistance.
+
+## Networking Errors
+
+If your task logs contain errors that indicate a networking issue, there could be a few possible causes.
+
+### Connections between Octopus Server and Tentacles
+
+Octopus Server communicates with Tentacles in either Listening mode or Polling mode. Both modes require different configuration. 
+
+A common problem is that traffic on the appropriate ports (10933 by default for Listening Tentacles) is not allowed by your firewall. If you are encountering problems with your connections, then your Task log might show messages that indicate a connection timing out, or a connection that was rejected by the remote host.
+
+See [Tentacle Communication Modes](docs/infrastructure/deployment-targets/windows-targets/tentacle-communication.md) for more information on configuring your Tentacles.
+
+#### Halibut
+
+The tool that manages connections between Octopus Server and your Tentacles is called Halibut. In order to discover more detailed information about the connections, it may be useful to [increase the log level for Halibut](docs/support/log-files.md#Logfiles-Changingloglevelshalibut). The same change to increase the log level for Halibut can also be made on [Tentacle](docs/support/log-files.md#Logfiles-Changingloglevelstentacle).
```

Comment:
Suggestion: link off to the Halibut readme to give people more detail if they want it.

```suggestion
The tool that manages connections between Octopus Server and your Tentacles is called [Halibut](https://github.com/OctopusDeploy/Halibut/). In order to discover more detailed information about the connections, it may be useful to [increase the log level for Halibut](docs/support/log-files.md#Logfiles-Changingloglevelshalibut). The same change to increase the log level for Halibut can also be made on [Tentacle](docs/support/log-files.md#Logfiles-Changingloglevelstentacle).
```